spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -318,37 +354,100 @@ protected void doStop() {
 
 		private final AckMode ackMode = getAckMode();
 
+		private final boolean isManualAck = this.ackMode.equals(AckMode.MANUAL);
+
+		private final boolean isManualImmediateAck = this.ackMode.equals(AckMode.MANUAL_IMMEDIATE)
+				|| this.ackMode.equals(AckMode.MANUAL_IMMEDIATE_SYNC);
+
+		private final boolean isAnyManualAck = this.isManualAck || this.isManualImmediateAck;
+
+		private final boolean isRecordAck = this.ackMode.equals(AckMode.RECORD);
+
 		private final boolean syncCommits = KafkaMessageListenerContainer.this.syncCommits;
 
-		private Thread consumerThread;
+		private final long pauseAfter = getPauseAfter();
+
+		private final boolean pauseEnabled = isPauseEnabled();
```

Comment:
We should always use the boolean - I propagated these properties to the LC to avoid referencing the outer class on every access - in other words for performance reasons. I am ok and completely concur with using a properties container, though. But, it should be a hard final reference in the consumer, again to avoid accessing the container instance:

    isPauseEnabled();

    LINENUMBER 616 L0
    ALOAD 0
    GETFIELD org/springframework/kafka/listener/KafkaMessageListenerContainer$ListenerConsumer.this$0 : Lorg/springframework/kafka/listener/KafkaMessageListenerContainer;
    INVOKEVIRTUAL org/springframework/kafka/listener/KafkaMessageListenerContainer.isPauseEnabled ()Z
    IFEQ L1

    this.pauseEnabled:

    LINENUMBER 616 L0
    ALOAD 0
    GETFIELD org/springframework/kafka/listener/KafkaMessageListenerContainer$ListenerConsumer.pauseEnabled : Z
    IFEQ L1